packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -107,12 +107,11 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
       lastColumnIndex,
     };
   }, [
-    apiRef,
-    containerWidth,
-    rootProps.autoHeight,
+    columnsMeta.positions,
+    rowsMeta.positions,
     disableVirtualization,
-    rowHeight,
     currentPage.rows.length,
+    containerWidth,
     visibleColumns.length,
   ]);
 
```

Comment:
Another finding doing #3387. If `rowsMeta.totalHeight` changes in a way where it causes the vertical scrollbar to be visible e.g. click a button to change the row height, the container width needs to be recalculated. I did this in https://github.com/mui-org/material-ui-x/pull/3387/files#diff-7fb7dbb1d7fe2c699d9fc5cfc5c6a56b2489af3b46dc1a2a8eaa3404575cacbdR136